File: frontend/src/types/iso20022.ts
export interface DomainInfo {
  id: string;
  name: string;
}

export interface MessageInfo {
  message_id: string;
  message_name: string;
  submitting_org: string;
  business_area: string;
  xsd_url: string | null;
}

export interface ParsedField {
  name: string;
  xsd_type: string;
  mapped_generator: string;
  min_occurs: number;
  max_occurs: string;
  documentation: string | null;
  enumeration_values: string[] | null;
  nested_fields: ParsedField[] | null;
}

export interface XsdParsedResponse {
  message_id: string;
  message_name: string;
  namespace: string | null;
  fields: ParsedField[];
}

File: backend/app/schemas/iso20022.py
from pydantic import BaseModel


class DomainInfo(BaseModel):
    id: str
    name: str


class MessageInfo(BaseModel):
    message_id: str
    message_name: str
    submitting_org: str
    business_area: str
    xsd_url: str | None = None


class ParsedField(BaseModel):
    name: str
    xsd_type: str
    mapped_generator: str
    min_occurs: int = 1
    max_occurs: str = "1"
    documentation: str | None = None
    enumeration_values: list[str] | None = None
    nested_fields: list[ParsedField] | None = None


class XsdParsedResponse(BaseModel):
    message_id: str
    message_name: str
    namespace: str | None = None
    fields: list[ParsedField]
